dhcpv4/bsdp.go

```diff
@@ -0,0 +1,328 @@
+// as the number of bytes read from the slice.
+func bootImageFromBytes(bytes []byte) (*BootImage, int, error) {
+	// If less than length of boot image ID and count, it's probably invalid.
+	if len(bytes) < 5 {
+		return nil, 0, fmt.Errorf("not enough bytes for BootImage")
+	}
+	imageID := bootImageIDFromBytes(bytes[:4])
+	nameLength := int(bytes[4])
+	if 5+nameLength > len(bytes) {
+		return nil, 0, fmt.Errorf("not enough bytes for BootImage")
+	}
+	name := string(bytes[5 : 5+nameLength])
+	return &BootImage{ID: imageID, Name: name}, 5 + nameLength, nil
+}
+
+// makeVendorClassIdentifier calls the sysctl syscall on macOS to get the
+// platform model.
+func makeVendorClassIdentifier() (string, error) {
+	// Fetch hardware model for class ID.
+	hwModel, err := syscall.Sysctl("hw.model")
+	if err != nil {
+		return "", err
+	}
+	vendorClassID := fmt.Sprintf("AAPLBSDPC/i386/%s", hwModel)
+	return vendorClassID, nil
+}
+
+// parseBootImagesFromBSDPOption parses data from the BSDPOptionBootImageList
+// option and returns a list of BootImages.
+func parseBootImagesFromBSDPOption(data []byte) ([]BootImage, error) {
+	// Should at least have the # bytes of boot images.
+	if len(data) < 4 {
+		return nil, fmt.Errorf("invalid length boot image list")
+	}
+
+	readByteCount := 0
+	start := data
+	var bootImages []BootImage
+	for {
+		bootImage, readBytes, err := bootImageFromBytes(start)
+		if err != nil {
+			return nil, err
+		}
+		bootImages = append(bootImages, *bootImage)
+		readByteCount += readBytes
+		if readByteCount+1 >= len(data) {
+			break
+		}
+		start = start[readByteCount:]
+	}
+
+	return bootImages, nil
+}
+
+// parseVendorOptionsFromOptions extracts the sub-options list of the vendor-
+// specific options from the larger DHCP options list.
+func parseVendorOptionsFromOptions(options []Option) []Option {
+	var vendorOpts []Option
+	var err error
+	for _, opt := range options {
+		if opt.Code == OptionVendorSpecificInformation {
+			vendorOpts, err = OptionsFromBytes(opt.Data)
+			if err != nil {
+				return []Option{}
+			}
+			break
+		}
+	}
+	return vendorOpts
+}
+
+// ParseBootImageListFromAck parses the list of boot images presented in the
+// ACK[LIST] packet and returns them as a list of BootImages.
+func ParseBootImageListFromAck(ack DHCPv4) ([]BootImage, error) {
+	var bootImages []BootImage
+	vendorOpts := parseVendorOptionsFromOptions(ack.options)
+	for _, opt := range vendorOpts {
+		if opt.Code == BSDPOptionBootImageList {
+			images, err := parseBootImagesFromBSDPOption(opt.Data)
+			if err != nil {
+				return nil, err
+			}
+			bootImages = append(bootImages, images...)
+		}
+	}
+
+	return bootImages, nil
+}
+
+// NewInformListForInterface creates a new INFORM packet for interface ifname
+// with configuration options specified by config.
+func NewInformListForInterface(iface string, replyPort uint16) (*DHCPv4, error) {
+	d, err := NewInformForInterface(iface /* needsBroadcast */, false)
```

Comment:
can you remove the `/* needsBroadcast */` comment if it's a leftover?

Reply:
I was going to leave it there to clarify what the `false` param is for (since go doesn't have kwargs :( )

Reply:
ah, so I guess we can steal from C++ here and make something like

```Go
    d, err := NewInformForInterface(iface, /* needsBroadcast = */ false)
```